File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
numLetters = 0
numWords = 1
numSetences = 0

for i in range(len(text)):
    if text[i].isalpha():
        numLetters = numLetters + 1
    elif text[i] == " ":
        numWords = numWords + 1
    elif text[i] in (".", "!", "?"):
        numSetences = numSetences + 1

grade = round(5.88 * numLetters / numWords - 29.6 * numSetences / numWords - 15.8)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade ", grade)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int numLetters = 0;
    int numWords = 0;
    int numSentences = 0;

    for (int i = 0; text[i] != '\0' ; i++)
    {
        //Count if is an alfabethic character
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            numLetters++;
        }
        //Count if is an space, or if it's the end of the text from input
        if (text[i] == ' ' || ((text[i] == '.' || text[i] == '!' || text[i] == '?') && text[i + 1] == '\0'))
        {
            numWords++;
        }
        //Count number of .!? characters in text
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            numSentences++;
        }
    }

    float index = 5.88 * (float) numLetters / numWords - 29.6 * (float) numSentences / numWords - 15.8;

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }
}